src/sidebar/services/load-annotations.js

```diff
@@ -16,53 +16,73 @@ export default function loadAnnotationsService(
   let searchClient = null;
 
   /**
-   * Load annotations for all URIs and groupId.
+   * Load all annotations that match `options` criteria
    *
-   * @param {string[]} uris
-   * @param {string} groupId
+   * @param {Object} options
+   *   @param {string} options.groupId
+   *   @param {string[]} [options.uris]
    */
-  function load(uris, groupId) {
+  function load(options) {
+    const { uris } = options;
     store.removeAnnotations(store.savedAnnotations());
 
     // Cancel previously running search client.
     if (searchClient) {
       searchClient.cancel();
     }
 
-    if (uris.length > 0) {
-      searchAndLoad(uris, groupId);
-
+    if (uris && uris.length > 0) {
       streamFilter.resetFilter().addClause('/uri', 'one_of', uris);
       streamer.setConfig('filter', { filter: streamFilter.getFilter() });
     }
+    searchAndLoad(options);
   }
 
-  function searchAndLoad(uris, groupId) {
+  /**
+   * @param {Object} options
+   *   @param {string[]} [options.uris]
+   *   @param {string} options.groupId
+   */
+  function searchAndLoad(options) {
+    const { groupId, uris } = options;
+
     searchClient = new SearchClient(api.search, {
       incremental: true,
       separateReplies: false,
     });
+
     searchClient.on('results', results => {
       if (results.length) {
         store.addAnnotations(results);
       }
     });
+
     searchClient.on('error', error => {
       console.error(error);
     });
+
     searchClient.on('end', () => {
       // Remove client as it's no longer active.
       searchClient = null;
 
-      store.frames().forEach(function (frame) {
-        if (0 <= uris.indexOf(frame.uri)) {
-          store.updateFrameAnnotationFetchStatus(frame.uri, true);
-        }
-      });
+      if (uris && uris.length > 0) {
+        store.frames().forEach(function (frame) {
+          if (0 <= uris.indexOf(frame.uri)) {
```

Comment:
```suggestion
          if (uris.indexOf(frame.uri) <= 0) {
```

Reply:
Interesting. Huh. This is inherited code, and I just spent a little time trying to track down what its intentions are. It first materializes in [this commit](https://github.com/hypothesis/client/commit/fd46c7d82348c6d28f0efc0d8356865105e4a976#diff-b85532614485654e161b1a77d2a893530b28d4ffabcb16d7117da685cfd9bb3d).

This code iterates through each `frame` object in the store. The conditional here will evaluate true if the `frame`'s `uri` property is either the 0th element in `uris` or is not present in `uris`. _This seems really weird_. Intuitively it doesn't make sense, but I'm nervous to try adjusting the logic here lest there be some sort of magic sauce I don't grok.

I _can_, however, flip the expression around as you suggest. I'm just...lost as to the purpose of this code and would like to get that sorted out for posterity.

There _is_ a test pertaining to the code here: https://github.com/hypothesis/client/commit/fd46c7d82348c6d28f0efc0d8356865105e4a976#diff-dbc0db35fdc8179fe9d2a754e50f85579192b1a77ca1430afae62c41931a8acdR187 but it doesn't really shed any light.

Reply:
👍🏼